test/Npgsql.Tests/DistributedTransactionTests.cs

```diff
@@ -377,6 +377,38 @@ public void Connection_reuse_race_chaining_transaction([Values(false, true)] boo
         }
     }
 
+    [Test, IssueLink("https://github.com/npgsql/npgsql/issues/5246")]
+    public void Transaction_complete_with_undisposed_connections()
+    {
+        using (var deleteTX = new TransactionScope())
+        {
+            var outerTX = Transaction.Current;
+            using (var copyTX = new TransactionScope(TransactionScopeOption.RequiresNew))
```

Comment:
nit: I don't think all these nested TransactionScopes should be necessary to repro the problem, so this should be simplifiable if you feel like it.

Reply:
Sure can do, just thought having a complete repro might help in the future,